src/rdkafka.h

```diff
@@ -2270,6 +2284,60 @@ typedef enum rd_kafka_cert_enc_t {
 } rd_kafka_cert_enc_t;
 
 
+/**
+ * @brief Sets a callback that provides a client certificate
+ *
+ * This callback, if set, is triggered from internal librdkafka threads
+ * upon connecting to a broker. If it returns a TLS certificate/key,
+ * this is used in preference to any value set with the `ssl.key.` and
+ * `ssl.certificate.` family of options.
+ *
+ * The pointers *certificate_buf_size and *key_buf_size initially contain the
+ * maximum amount of bytes that can be written to *certificate_buf and *key_buf;
+ * the callback is expected to overwrite these values with the actual length of
+ * the certificate/key data it wrote (which should be no larger than the initial
+ * value of the buffer size)
+ *
+ * If the callback is able to provide a client certificate/key to librdkafka,
+ * it should:
+ *   - Return RD_KAFKA_CERT_FETCH_OK,
+ *   - write a certificate to certificate_buf,
+ *   - write the format of that certificate to certificate_format,m
+ *   - write the size of the certificate to certificate_buf_size,
+ *   - write a key to key_buf,
+ *   - write the format of that key to key_format
+ *   - write the size of the key to key_buf_size,
+ *
+ * Otherwise, if the provided buffers are too small (i.e *certificate_buf_size
+ * or *key_buf_size are too small), then the callback should:
+ *   - Return RD_KAFKA_CERT_FETCH_MORE_BUFFER,
+ *   - write the desired buffer size to *buf_size
+ * Note that the first time rd_kafka_conf_set_ssl_cert_fetch_cb is called, it is
+ * called with zero for *buf_size - i.e. you will always need to initially tell
+ * librdkafka how large your buffer needs to be.
+ *
+ * If the callback is not able to provide a cert/key pair, it should simply
+ * return RD_KAFKA_CERT_FETCH_NONE.
+ *
+ * @returns RD_KAFKA_CONF_OK if SSL is supported in this build, else
+ *          RD_KAFKA_CONF_INVALID.
+ *
+ * @warning This callback will be called from internal librdkafka threads.
+ * */
+RD_EXPORT
+rd_kafka_conf_res_t rd_kafka_conf_set_ssl_cert_fetch_cb(
+        rd_kafka_conf_t *conf,
+        int (*ssl_cert_fetch_cb) (rd_kafka_t *rk,
+                                  const char *broker_name,
+                                  int32_t broker_id,
+                                  char *buf, size_t *buf_size,
+                                  char **leaf_cert, size_t *leaf_cert_size,
+                                  char **pkey, size_t *pkey_size,
+                                  char *chain_certs[16], size_t chain_cert_sizes[16],
+                                  rd_kafka_cert_enc_t *format,
```

Comment:
add a p suffix to all return-type pointers, makes them easier to differentiate in the code.
E.g.: `*buf_sizep`, `*formatp`